src/Abstractions/ITableStorage.cs

```diff
@@ -80,5 +81,17 @@ public interface ITableStorage
         /// <param name="table">The table to remove the entity from</param>
         /// <param name="etag">The etag of the entity to remove; used to prevent stale data deletes</param>
         Task DeleteEntity(string id, string table, string etag = "*");
+
+        /// <summary>
+        /// Deletes entities in batches in a given table
+        /// </summary>
+        /// <param name="tableName">The name of the table</param>
+        /// <param name="query">TableQuery</param>
+        /// <param name="continuationToken">TableContinuationToken</param>
+        /// <param name="cancellationToken">CancellationToken</param>
+        /// <returns>Task</returns>
+        Task DeleteInBatchAsync(string tableName, TableQuery<DynamicTableEntity> query,
```

Comment:
Do not leak Azure-specific resources (TableQuery, TableContinuationToken) from this library (per above)